### PR TITLE
Adds a custom model and item box icon for Mega TCs.

### PR DESCRIPTION
**Pull request by Zachruff**
Model by Atlas
Icon by professorchaos

To do: add two breff effect files that accompany these visuals.

### CODE REVIEW

File: payload/game/system/DVDArchive.cc [hunk 1]
```diff
@@ -17,6 +21,18 @@ void *DVDArchive::getFile(const char *path, size_t *size) {
...
+    auto* saveManager = System::SaveManager::Instance();
+    auto setting = saveManager->getSetting<SP::ClientSettings::Setting::VSMegaClouds>();
+    if (setting == SP::ClientSettings::VSMegaClouds::Enable) {
+        if (!strcmp(path, "kumo.brres")) {
+            return REPLACED(getFile)("MegaTC.brres", size);
+        }
+        /*
+            if (!strcmp(path, "fm_item_pikakumo.tpl")) {
+                return REPLACED(getFile)("MegaTC_icon.tpl", size);
+            } */
```
Comment (vabold) [inline payload/game/system/DVDArchive.cc:33]: Remove comment.

File: payload/game/util/Registry.cc [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+            auto* saveManager = System::SaveManager::Instance();
+            auto setting = saveManager->getSetting<SP::ClientSettings::Setting::VSMegaClouds>();
+
+            // If the item is a thundercloud, and the option for mega tcs is enabled, return the mega tc
+            // pane, otehrwise go back to original function
+            if (itemId == 0x0E) {
+                if (setting == SP::ClientSettings::VSMegaClouds::Enable) {
+                    return "mega_c";
+                }
+            }
```
Comment (vabold) [inline payload/game/util/Registry.cc:17]: Merge conditions.

File: payload/game/util/Registry.cc [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+#include "game/system/SaveManager.hh"
+
+    namespace Registry {
+
+        const char* getItemPane(u32 itemId, u32 count) {
+            auto* saveManager = System::SaveManager::Instance();
+            auto setting = saveManager->getSetting<SP::ClientSettings::Setting::VSMegaClouds>();
+
+            // If the item is a thundercloud, and the option for mega tcs is enabled, return the mega tc
+            // pane, otehrwise go back to original function
```
Comment (vabold) [inline payload/game/util/Registry.cc:12]: "otehrwise"

I don't think we need a comment here, the code is fairly self-explanatory. Maybe "Return mega TC pane".

File: payload/game/util/Registry.cc [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+                if (setting == SP::ClientSettings::VSMegaClouds::Enable) {
+                    return "mega_c";
+                }
+            }
+            
+            return REPLACED(getItemPane)(itemId, count);
+            
+        }
+
+    } // namespace Registry
```
Comment (vabold) [inline payload/game/util/Registry.cc:23]: Newline.

File: payload/game/util/Registry.cc [hunk 5]
```diff
@@ -0,0 +1,23 @@
+#include "Registry.hh"
+
+#include "game/system/SaveManager.hh"
+
+    namespace Registry {
+
+        const char* getItemPane(u32 itemId, u32 count) {
```
Comment (vabold) [inline payload/game/util/Registry.cc:7]: Please run clang-format.

File: payload/game/util/Registry.hh [hunk 6]
```diff
@@ -10,4 +10,9 @@ u32 GetCharacterMessageId(u32 characterId, bool resolveMiiNames);
 
 const char *GetCharacterPane(u32 characterId);
 
+const char* REPLACED(getItemPane)(u32 itemId, u32 count);
+REPLACE const char* getItemPane(u32 itemId, u32 count);
+
+const char* GetItemPane(u32 itemId, u32 count);
```
Comment (vabold) [inline payload/game/util/Registry.hh:16]: We do not need `GetItemPane`.

STYLE: `const char* getItemPane` should be `const char *getItemPane`.

**Review by vabold — CHANGES_REQUESTED** [review]

File: payload/game/util/Registry.hh [hunk 7]
```diff
@@ -10,4 +10,7 @@ u32 GetCharacterMessageId(u32 characterId, bool resolveMiiNames);
 
 const char *GetCharacterPane(u32 characterId);
 
+const char *REPLACED(getItemPane)(u32 itemId, u32 count);
```
Comment (riidefi) [inline payload/game/util/Registry.hh:13]: (Optional) Capitalize first letter to match other functions in this file.

**Review by stblr — APPROVED** [review]

**Pull request merged**